### ISSUE TITLE
@keyframes fail to work with rsg-components/Styled

### ISSUE THREAD

**Issue by josephrexme** [styleguidist/react-styleguidist#1695]
When I try to use a @keyframe animation following this [cookbook example](https://github.com/styleguidist/react-styleguidist/blob/e662394aea164c78f11408cddca4bcfba2998cc2/docs/Cookbook.md#how-to-change-styles-of-a-style-guide), I get:

```
TypeError: container.addRule(...).addRule is not a function
    in Styled(StyleGuide) (created by StyleGuide)
    in StyleGuide
```
At first I tried to do this in the component I was overriding and got that error, then I tried to do it from the style section of the configuration and got the same. Here's what the code looks like:

```js
content: {
      opacity: 0,
      animation: 'fade .3s ease .5s forwards',
      '@keyframes fade': {
        to: { opacity: 1 }
      },
}
```

**Comment by sapegin** [comment]
Feel free to send a pull request with a fix 👾

**Comment by josephrexme** [comment]
I figured out a solution to this and it seem this might not be a bug afterall (as far as react-styleguidist) but just something that needs updated documentation. I had created [this codesandbox](https://codesandbox.io/s/rough-fire-ytjwp?file=/src/App.js:359-371) to reproduce the issue with JSS and the 2 things I learned from doing that are:

1) From JSS 10.0.0 (what react-styleguidist is on as of now), [keyframe names are scoped](https://github.com/cssinjs/jss/issues/1120) so we'll have to use the animation as:

```js
animation: '$fade .3s ease .5s forwards'
```

2) Animations wouldn't work from a nested property. They have to be at the root. So rewriting that whole animation this way will work:

```js
const styles = () = ({
  content: {
    opacity: 0,
    animation: '$fade .3s ease .5s forwards',
  },
  '@keyframes fade': {
    to: { opacity: 1 }
  },
})
```
I'll still make a PR to add this to the cookbook section of the documentation before I close this. But maybe it doesn't have to be labeled as bug anymore?
